python/ray/data/impl/plan.py

```diff
@@ -149,6 +296,25 @@ def __init__(
         self.supports_block_udf = supports_block_udf
         self.block_udf = block_udf
 
+    def can_fuse(self, prev: Stage):
+        context = DatasetContext.get_current()
+        if not context.optimize_fuse_shuffle_stages:
+            return False
+        if not self.supports_block_udf:
+            return False
+        if not isinstance(prev, OneToOneStage):
+            return False
+        if prev.compute != "tasks":
+            return False
+        if prev.ray_remote_args != {}:
```

Comment:
Nit: More idiomatic to check if the dict is falsey.
```suggestion
        if not prev.ray_remote_args:
```